src/schema/posts.ts

```diff
@@ -741,11 +742,29 @@ export const resolvers: IResolvers<any, Context> = {
       return { _: true };
     },
     deletePost: async (
-      source,
+      _,
       { id }: { id: string },
       ctx: Context,
     ): Promise<GQLEmptyResponse> => {
-      await ctx.getRepository(Post).update({ id }, { deleted: true });
+      if (ctx.roles.includes(Roles.Moderator)) {
+        await ctx.getRepository(Post).update({ id }, { deleted: true });
+        return { _: true };
+      }
+
+      await ctx.con.transaction(async (manager) => {
+        const repo = manager.getRepository(Post);
+        const post = await repo.findOneBy({ id });
+        if (post.type === PostType.Share) {
```

Comment:
I'm not sure if it's the right condition or whether we should check if it's a squad post. Let's keep it for now like this